src/vignette/storage/static_assets.clj

```diff
@@ -22,23 +22,27 @@
     (fs/file-exists? to))
   (->response-object [this] (file-stream this)))
 
-(defn create-async-response-stored-object [url]
-  (let [response @(http/get url {:as :stream})]
-    (if (-> response :status (= 200))
-      (->AsyncResponseStoredObject response))))
+(defn create-async-response-stored-object [static-image-get image-review-get]
+  (let [static-image-response @static-image-get]
+    (if (and (-> static-image-response :status (= 200)) (-> @image-review-get :status (= 200)))
+      (->AsyncResponseStoredObject static-image-response))))
```

Comment:
Minor: it would be nice to have a method that will either dereference the future or pass through a map so that you don't have to wrap the response in a `future` in cases where it's not needed e.g. the tests. Something like:

```clojure
(def deref-request [f]
   (if (future? f)
       (deref f)
       f))
```